transport/http_util.go

```diff
@@ -159,6 +161,8 @@ func newHPACKDecoder() *hpackDecoder {
 			}
 		case ":path":
 			d.state.method = f.Value
+		case "content-type":
```

Comment:
On Sun, Mar 22, 2015 at 5:53 PM, David Symonds <notifications@github.com>
wrote:

> In transport/http_util.go
> <https://github.com/grpc/grpc-go/pull/126#discussion_r26909794>:
>
> > @@ -159,6 +161,8 @@ func newHPACKDecoder() *hpackDecoder {
> >  			}
> >  		case ":path":
> >  			d.state.method = f.Value
> > +		case "content-type":
>
> The design for this feature was such that the client explicitly specifies
> to use a non-standard codec. It'd be nice just from the symmetry
> perspective to have the server be similarly explicit. I don't think it's a
> particularly necessary goal that a server can automatically switch between
> wire formats, but maybe you're thinking about that?
>
That is one thing I thought about it. But I also agree that it is not
particularly necessary. My real question is that if we want to adopt
symmetry where and how to specify it on server side. If you have a good
solution to my question, I am okay to switch to it.

> If we're going for some arbitrary string-matching scheme here, the codec
> package (or wherever that code lives) will probably need a registration
> mechanism. We can't bake in knowledge about every possible codec.
>
Thought about it too. But notice that this is on the fast path so that I do
not want to introduce a table lookup for it.

> The use of the codec's String() result in Content-Type needs explicit
> documentation in the Codec interface definition.
>
acked.

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc-go/pull/126/files#r26909794>.
>